src/dataframe/dataframe.jl

```diff
@@ -502,34 +516,46 @@ end
         throw(BoundsError(df, (row_inds, col_inds)))
     end
     selected_columns = index(df)[col_inds]
-    # Computing integer indices once for all columns is faster
-    selected_rows = T === Bool ? findall(row_inds) : row_inds
-    new_columns = AbstractVector[dv[selected_rows] for dv in _columns(df)[selected_columns]]
+    old_columns = _columns(df)[selected_columns]
+    new_columns = Vector{AbstractVector}(undef, length(old_columns))
+
+    if length(new_columns) == 1
+        new_columns[1] = only(old_columns)[row_inds]
+    elseif length(new_columns) > 1
+        # Computing integer indices once for all columns is faster
+        selected_rows = T === Bool ? findall(row_inds) : row_inds
```

Comment:
@nalimilan - regarding Polars this is the place where the `findall` trick for leading 1s or 0s in `BitVector` could potentially be used, but I think we should not do it and delegate such optimizations to JuliaBase.

Reply:
Yes better discuss this in Base first.

Even if Base implements an optimized `getindex` method for `BitVector`, what's special here is that we repeatedly use the same indices on multiple columns. So `getindex` on `BitVector` would have to recompute the sum again and again. Ideally we would compute it only once.

Reply:
Yes - I had that in mind, but maybe there are some additional optimizations that can be implemented. These things are tricky :)